src/Form/Input.js

```diff
@@ -46,7 +60,19 @@ const StyledInput = createComponent({
 
     ${isFloating &&
       css`
-        padding-bottom: 0;
+        line-height: calc(${theme.fontSizes[size]}px * 1.2);
+        padding-top: calc(${theme.fontSizes[size]}px * 1.2);
+        padding-bottom: 0px;
+      `};
+
+    ${hasLeftIcon &&
+      css`
+        padding-left: ${leftIconSize + 12}px;
+      `};
+
+    ${hasRightIcon &&
+      css`
+        padding-right: ${rightIconSize + 32}px;
```

Comment:
An alternative to passing down the icon props as already discussed [here](https://github.com/heydoctor/molekule/pull/49#discussion_r309821111), would be to pass down the icon component as a `rightIcon` or `leftIcon` prop (for example): `rightIcon={<Icon name="..." color="..." size={#}/>}` and its props could be accessed via:
```
${rightIcon &&
  css`
    padding-right: ${rightIcon.props.size + 32}px;
...
```

Reply:
@cehsu I've made some changes to try this example you've given. Let me know what you think 👍

Reply:
I'm also curious about some other possible ways, what do you think:
`<Input leftIcon="alert-circle" styles={{ leftIcon: { color: 'primary' } }} />`
or
`<Input leftIcon={{ name: 'alert-circle', size: 16, color: 'primary' }} />`

Reply:
My thinking for removing the hasLeftIcon, and the hasRightIcon styles is that the padding amounts might be non-standard (i.e. not always 12px). Instead users could specify subcomponent styles via a styles prop (as mentioned in the overridable styles section):
```
<Input styles={{Input: { ... }}}/>
```

If we pass the Icon components in, rather than the icon props we might be able to simplify this component (so it doesn't have to manage the Icon implementation), and also make it more flexible. How does that sound @mathewmorris?

Reply:
@cehsu That sounds great. I think keeping the Input out of the loop of Icon component is a good thing. Just render what I give you. 👍

Reply:
Sounds good, the approach of having the input handle the icon implementation also has precedent, but would probably require additional variables such as how [Semantic UI](https://react.semantic-ui.com/elements/input/#variations-icon) handles this via an iconMargin variable:
```
.ui.icon.input > input {
  padding-right: @iconMargin !important;
}
```
For now, an approach like [react-select](https://react-select.com/home#custom-styles), where custom styles get passed in seems like a closer delta.